<?xml version="1.0" encoding="UTF-8" ?>

<launch>  

  <!-- Arguments and parameters --> 
  <arg name="robot_namespace" />
  <arg name="robot_base_frame" default = "robot1/base_footprint"/>
  <!--arg name="move_forward_only" default="false"/-->

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <!--param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" /-->

    <rosparam file="$(find first_robot_navigation)/param/costmap_common_params_robot1.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find first_robot_navigation)/param/costmap_common_params_robot1.yaml" command="load" ns="local_costmap"/>
    <rosparam file="$(find first_robot_navigation)/param/local_costmap_params_robot1.yaml" command="load"/>
    <rosparam file="$(find first_robot_navigation)/param/global_costmap_params_robot1.yaml" command="load"/>
    <rosparam file="$(find first_robot_navigation)/param/base_local_planner_params_robot1.yaml" command="load"/>
    <rosparam file="$(find first_robot_navigation)/param/move_base_params.yaml" command="load"/>

    <!--rosparam file="$(find first_robot_navigation)/param/dwa_local_planner_params_burger.yaml" command="load" /-->

    <!-- Set tf_prefix for frames explicity, overwriting defaults 
    <param name="global_costmap/scan/sensor_frame" value="$(arg robot_namespace)/base_scan"/>
    <param name="global_costmap/obstacle_layer/scan/sensor_frame" value="$(arg robot_namespace)/base_scan"/>
    <param name="global_costmap/global_frame" value="map"/>
    <param name="global_costmap/robot_base_frame" value="$(arg robot_namespace)/base_footprint"/>
    <param name="local_costmap/scan/sensor_frame" value="$(arg robot_namespace)/base_scan"/>
    <param name="local_costmap/obstacle_layer/scan/sensor_frame" value="$(arg robot_namespace)/base_scan"/>
    <param name="local_costmap/global_frame" value="$(arg robot_namespace)/odom"/>
    <param name="local_costmap/robot_base_frame" value="$(arg robot_namespace)/base_footprint"/> -->

    <remap from="robot_base_frame" to="robot1/base_link"/>
    <remap from="cmd_vel" to="robot1/cmd_vel"/>
    <remap from="odom" to="robot1/odom"/>
    <remap from="/map" to="robot1/map"/>
    <!--remap from="map" to="/map_merged_2"/-->

  </node> 

</launch>


